hardware_interface/source/eps.c

```diff
@@ -141,16 +141,32 @@ void EPS_getHK(eps_instantaneous_telemetry_t* telembuf) {
 }
 
 eps_mode_e get_eps_batt_mode() {
+    eps_refresh_instantaneous_telemetry();
     eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
     return (eps_mode_e) eps.battMode;
 }
 
-int8_t eps_get_pwr_chnl(uint8_t pwr_chnl_port){
-
+/* Gets the status of the power channel */
+uint8_t eps_get_pwr_chnl(uint8_t pwr_chnl_port){
+    eps_refresh_instantaneous_telemetry();
+    eps_instantaneous_telemetry_t eps = get_eps_instantaneous_telemetry();
+    uint32_t outputStatus = eps.outputStatus; // a codeword that has the status of all channels
+    uint8_t pwr_chnl_status = (uint8_t) (outputStatus >> (pwr_chnl_port - 1)) & 1; // chnl_port : 1-18
+    return pwr_chnl_status;

```

Comment:
There is no problem here. But if I were you, I just do `return (uint8_t) (eps.outputStatus >> (pwr_chnl_port -1)) & 1;`  😆

Reply:
Thanks. This way it's more obvious what the function is returning.